src/js/player.js

```diff
@@ -2344,15 +2350,19 @@ class Player extends Component {
         return;
       }
 
-      // At this point, we've found a compatible tech for the middleware source,
-      // so we can tell any subscribers that the player is ready again.
-      this.triggerReady();
+      // Notify middlewares of a potentially new tech.
+      middleware.setTech(mws, this.tech_);
 
-      // Cache the source URL.
-      this.cache_.src = middlewareSource.src;
+      this.tech_.ready(() => {
 
-      // Notify middlewares of a new tech.
-      middleware.setTech(mws, this.tech_);
+        // Cache the source URL.
+        this.cache_.src = middlewareSource.src;
+
+        // At this point, we've found a compatible tech and source and set the
+        // source both on the tech and in the internal cache. It's safe to tell
+        // any subscribers that the player is ready again.
+        this.triggerReady();
```

Comment:
do we even need to do this? Should we be getting ready from the tech for the source change in the first place?

Reply:
Well, the whole idea here is that tech ready != player ready - this is the latter. The tech is ready before the source change is complete. Maybe I'm not following the line of questions.

Reply:
Well, when the player gets the tech's `ready` event, it already triggers ready itself: https://github.com/videojs/video.js/blob/c7ad7b31f0d3a15a8d6cf82dfaf0033264fadd6b/src/js/player.js#L1055-L1061
So, we would end up getting two `ready` events for the same source unless we change the way that `handleTechReady` functions.